core/src/main/java/org/hisp/dhis/android/core/sms/data/LocalDbRepositoryImpl.java

```diff
@@ -32,10 +41,16 @@
 
     public LocalDbRepositoryImpl(Context ctx,
```

Comment:
I would recommend you here to add the @Inject annotation and in SmsDIModule: 

```
@Provides
    LocalDbRepository localDbRepository(LocalDbRepositoryImpl) {
        return impl;
    }
```

So you don't need to add the dependencies in both files.